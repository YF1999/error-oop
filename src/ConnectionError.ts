import { ErrorOptions, ConnectionErrorMessageProps, ConnectionErrorProps } from './CommonTypes';
import { _Error } from './ExtendedNodeError';

export class _ConnectionError extends _Error {
    public constructor(
        props: ConnectionErrorProps,
        options: ErrorOptions<ConnectionErrorMessageProps>,
    ) {
        super(props, options);
    }
}

/**
 * Applicable when an error occurs on a connection.
 */
export class ConnectionError extends _ConnectionError {
    public constructor();
    /**
     * @param message The error message that explains the reason for this error.
     */
    public constructor(message: string);
    /**
     * @param message The error message that explains the reason for this error.
     * @param innerError The error that is the cause of the current error. Stack trace will be
     * appended.
     */
    public constructor(message: string, innerError: Error);

    public constructor(message: string = '', innerError?: Error) {
        super({ message, innerError }, {});
    }
}
