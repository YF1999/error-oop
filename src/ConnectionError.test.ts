import { ConnectionError } from './ConnectionError';

standardTestError(ConnectionError, 'ConnectionError', 'error');
