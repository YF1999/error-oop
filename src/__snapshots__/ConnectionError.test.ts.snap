// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test \`ConnectionError\` own keys 1`] = `
Array [
  "stack",
]
`;

exports[`Test \`ConnectionError\` own keys 2`] = `
Array [
  "stack",
  "message",
]
`;

exports[`Test \`ConnectionError\` own keys 3`] = `
Array [
  "stack",
  "message",
]
`;

exports[`Test \`ConnectionError\` own keys 4`] = `
Array [
  "stack",
  "message",
]
`;
